### ISSUE TITLE
[Request]: Manual Car Transmission

### ISSUE THREAD

**Issue by Rxann** [Yimura/YimMenu#1237]
### Solution

Hotkeys to switch up and down gears(like a manual transmission)

### Reason

it would be cool & i dont think its in the game

**Comment by Yimura** [comment]
It's possible with the new fields I added to the CVehicle class on the GTAV-Classes repo.

Still thinking on how we'll implement this because it needs to obviously work for both controller and keyboard players in a convenient way.

There's also this mod: https://github.com/E66666666/GTAVManualTransmission

**Comment by Rxann** [comment]
Okay, I'll take a look at the classes repo

We could have it so you press like x on ps4 controller then use the arrow keys to switch gears.

**Comment by Yimura** [comment]
That sounds annoying, I'd probably make it more accessible like using X and square to shift up and down (B and Y for Xbox controllers)

**Comment by Rxann** [comment]
Okay, That makes a lot more sense. I may start experimenting with it to get a draft

**Comment by Rxann** [comment]
So, I have looked at some code in the repo, and I think we could base the code off of the turn signal looped command.

**Comment by Yimura** [comment]
Nah the code of that feature is bad, I'll do it some day. I haven't added it myself yet because I want to do it properly.

**Comment by TheGreenBandit** [comment]
i have no idea about the control but some simple code like this should work,

if(m_current_gear != m_top_gear && control)
	m_current_gear++;
if(m_current_gear != -1 && control)
	m_current_gear--

**Comment by Rxann** [comment]
I believe that the gear is frame based and the command would need to be looped.

**Comment by Yimura** [comment]
> I believe that the gear is frame based and the command would need to be looped.

Yes just open cheat engine, find the address in memory for the current gear of the vehicle you’re driving and freeze the value.
You’ll see that it will behave sorta like a manual transmission car if you were to put it in a high gear and try to start from a standstill.

**Comment by Rxann** [comment]
@Yimura is this coming anytime soon?

**Comment by TheGreenBandit** [comment]
I tried using them awhile ago and setting them doesnt work, i think its just for the display in the car.

**Comment by Rxann** [comment]
> I tried using them awhile ago and setting them doesnt work, i think its just for the display in the car.

I may look into what the mod does for manual transmission

**Comment by Yimura** [comment]
> I tried using them awhile ago and setting them doesnt work, i think its just for the display in the car.

You have to keep looping the value to force a gear, it worked when I tried it.

**Comment by Rxann** [comment]
> > I tried using them awhile ago and setting them doesnt work, i think its just for the display in the car.
> 
> 
> 
> You have to keep looping the value to force a gear, it worked when I tried it.

so it's easy to implement then?
